openedx/openedx-platform
# Consolidate two different implementations of serializing an XBlock to OLX

## Description

Ticket: https://github.com/openedx/modular-learning/issues/54

This is just a refactor to make some code that I've been working on more DRY. i.e. it's removing some minor technical debt by consolidating two very similar classes that are used to serialize individual XBlocks to OLX - one that uses "normal" OLX and one that uses a blockstore-specific variant.

Or at least, that was the plan - I actually found some minor inconsistencies in the API when adding more tests for the new combined API, so it does include some bugfixes too. Mainly, it now always includes `url_name` in the exported OLX except when in "blockstore mode"; before this it was inconsistent.

## Supporting information

See https://github.com/openedx/edx-platform/pull/31904 which introduced the duplication. I split this fix out into a separate PR so that the refactor can be evaluated on its own.

## Testing instructions

Check that the test cases provide coverage and the copy-paste feature is still working.

## Deadline

None

## Other information

Private ref. MNG-3636
